Kudu.Core/Jobs/JobsFileWatcher.cs

```diff
@@ -139,9 +140,12 @@ private void StartWatcher(object state)
 
         private void OnError(object sender, ErrorEventArgs e)
         {
-            Exception ex = e.GetException();
-            _traceFactory.GetTracer().TraceError(ex.ToString());
-            ResetWatcher();
+            HostingEnvironment.QueueBackgroundWorkItem(cancellationToken =>
```

Comment:
Please add comment here explaining why we're doing this. Other than that, looks good!